File: app/CausalityCSharp/CausalityLibrary/Type/Action.cs
using System;

namespace CausalityLibrary.Type
{
    [Serializable]
    public class Action
    {
        public SerialNumber SerialNumber
        { get; }

        public string Description
        { get; }

        public string DefaultValue
        { get; }

        public string ActorName
        { get; }

        public bool IsPerformed
        { get; set; }

        public Action(SerialNumber serialNumber, string description,
            string actorName, string defaultValue = null)
        {
            SerialNumber = serialNumber;
            Description = description;
            DefaultValue = defaultValue;
            ActorName = actorName;
            IsPerformed = false;
        }

        // Don't instantiate with default constructor!
        internal Action() { }
    }
}

File: app/CausalityJava/src/com/inha/dsp/causality/type/Action.java
package com.inha.dsp.causality.type;

import java.io.Serializable;

public class Action implements Serializable {
    private SerialNumber serialNumber;
    private String description;
    private String defaultValue;
    private String actorName;
    public boolean IsPerformed;

    public SerialNumber getSerialNumber() {
        return serialNumber;
    }
    public String getDescription() {
        return description;
    }
    public String getDefaultValue() {
        return defaultValue;
    }
    public String getActorName() {
        return actorName;
    }

    public Action(SerialNumber sn, String desc, String aname, String dvalue)
    {
        serialNumber = sn;
        description = desc;
        actorName = aname;
        defaultValue = dvalue;
        IsPerformed = false;
    }

    // Don't instantiate with default constructor!
    Action() {}
}
